clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -150,15 +151,33 @@ public void testCommitAsync_UserSuppliedCallback() {
     }
 
     @Test
-    @SuppressWarnings("unchecked")
     public void testCommitted() {
         Set<TopicPartition> mockTopicPartitions = mockTopicPartitionOffset().keySet();
-        mockConstruction(OffsetFetchApplicationEvent.class, (mock, ctx) -> {
-            when(mock.complete(any())).thenReturn(new HashMap<>());
-        });
-        consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
-        assertDoesNotThrow(() -> consumer.committed(mockTopicPartitions, Duration.ofMillis(1)));
-        verify(eventHandler).add(ArgumentMatchers.isA(OffsetFetchApplicationEvent.class));
+        CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> committedFuture = new CompletableFuture<>();
+        try (MockedConstruction<OffsetFetchApplicationEvent> mockConstruction =
```

Comment:
`mockConstruction` is unused. Also, why do we wrap under `try`? Do we expect `mockConstruction` to be closed in the end?

Reply:
We need to close the mockConstruction object in the end because it is a thread-lock controller.  If we don't close it, it would remain active and causes undesirable behavior for other tests.  I'm using mockConstruction to mock the construction of the event object, are there better ways to mimic/mock the behavior the event construction and method calls there?